### PR TITLE
Replace shortNumberFormat with renderShortNumber

### PR DESCRIPTION
**Pull request by brawaru**
This PR introduces new number utility function - `renderShortNumber`. It should work almost the same way as original `shortNumberFormat` function, though it also localizes units and accepts one more parameter - `customRenderer`.

`customRenderer` is a function that takes rendered short formatted number and also *ready-to-pluralize* number to format display result accordingly. *Ready-to-pluralize* number allows to correctly select plural for short numbers, respecting thousands and other short units.

Issue #12451 accurately describes the issue with using raw numbers when replacing counter with short version. In short words, it doesn't work with languages such as Russian, that require different plurals, according to the unit number was shortened to.

This is probably not ideal solution and math should be revised later.

All previous of shortNumberFormat were replaced with new function, and it became unused, therefore was removed to avoid misleading.

### :warning: Beware!

This PR introduces major change to the codebase that touches multiple files at the same time and created by the person that doesn't have much experience in Mastodon's codebase. It should be carefully reviewed and tested before merging to avoid breakages. *Phew.*

### CODE REVIEW

**Comment by Sasha-Sorokin** [comment]
Now waiting for CI to slap me in da face…

File: app/javascript/mastodon/components/common_counter.js
Comment (Sasha-Sorokin) [inline app/javascript/mastodon/components/common_counter.js:1]: This annotation used to enable TypeScript checks in the JavaScript files. In JavaScript annotations are taken from JSDocs. Enabling TypeScript checks also allows for quick renaming and other helpful functions. Probably harmless by itself, when TS is not available.

**Comment by Sasha-Sorokin** [comment]
Made a little playground here: https://codesandbox.io/s/compact-notation-fix-testing-8b7hg.

Only English/Russian at the moment, hope to get more translations sooner to check it.

**Review by Gargron — APPROVED** [review]

**Comment by Gargron** [comment]
@Sasha-Sorokin Why back to draft?

**Comment by Sasha-Sorokin** [comment]
@Gargron really want to check with other translators now, like someone speaking Polish or Arabic, they have not less weird plural systems than Russian has. If this won't work, then I'm not sure what to do, really hate making any hardcode exceptions.

**Comment by Sasha-Sorokin** [comment]
Imagine seeing [all this](https://gist.github.com/Sasha-Sorokin/43264b240795385ef6417956c274d23d) green\*:

<p align=center>Screenshot of console output with lots of green strings about passed tests for locales</p>

*\* there are still a few fails for Cyrillic Bosnian, but does it matter?*

In summary, a few days ago I discovered that ICU4J Java library allows to do, what we want to do here — there, when you format the number, you given a class, that you can pass to select plural function, and it should give you correct plural for compact notation.

I made it to create me a JSON file with all the supported by CLDR locales and a few tests for numbers: how it will look when formatted using exact same parameters, and what would passing formatted number to plural rules selector give. This JSON file I then used for testing in Node.js with same pluralReady function, if it will return the correct category.

So, turns out my code works pretty well with mostly no fails and the only warnings are CLDR data inconsistencies (either ICU4J or Node.js uses old data), but this doesn't matter. Now I'm pretty confident this code would work, but I think I can further improve it.

**Comment by Sasha-Sorokin** [comment]
:pray: Okay, this works (at least on my developer instance). I'll squash and mark it as ready for review.

**Review by Gargron — APPROVED** [review]

**Comment by Sasha-Sorokin** [comment]
@Gargron by the way, is this okay that `yarn run manage:translations` wants to turn half the translations into the dust? I wanted to include new keys so I can immediately translate on Crowdin after merge, but found out that there are too much changes and some of them really unjustified (like I don't remember we removed polls, hotkeys and stuff, did we?). Can you check later? Thanks 😇

**Pull request merged**
